lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -497,6 +497,37 @@ LogicalResult LowerLayersPass::runOnModuleBody(FModuleOp moduleOp,
         continue;
       }
 
+      // If this operation has non-passive operands defined outside the
+      // layerblock, then it may result in a lowering that creates output ports
+      // on the module.  This is illegal in the FIRRTL spec since no layerblock
+      // may write to values outside it.  Try to fix this up if possible.
+      if (llvm::any_of(op.getOperands(), [&](Value a) {
```

Comment:
Nit: could probably check for the ops first to avoid scanning all operands but shrug.

Reply:
(I don't need this addressed to land)

Reply:
Definitely better. Changed to do this in: https://github.com/llvm/circt/pull/7464/commits/d540ee6b412b2c578df0c50c5024b73cb0f4c12c

Reply:
Good suggestion. Addressed in the fixup: https://github.com/llvm/circt/pull/7464/commits/d540ee6b412b2c578df0c50c5024b73cb0f4c12c